src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -536,10 +569,11 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             VariableState initialState,
             ImmutableDictionary<BoundExpression, (NullabilityInfo, TypeSymbol)>.Builder analyzedNullabilityMapOpt,
             SnapshotManager.Builder snapshotBuilderOpt,
-            ArrayBuilder<(BoundReturnStatement, TypeWithAnnotations)> returnTypesOpt)
+            ArrayBuilder<(BoundReturnStatement, TypeWithAnnotations)> returnTypesOpt,
+            NullableWalker restoredWalker = null)
```

Comment:
>NullableWalker restoredWalker = null [](start = 12, length = 36)

Is this used anywhere?

Reply:
Yes, `AnalyzeAndRewriteSpeculation` passes in the restored nullable walker from a snapshot.

---
In reply to: [295514274](https://github.com/dotnet/roslyn/pull/36563#discussion_r295514274) [](ancestors = 295514274)

Reply:
> Yes, AnalyzeAndRewriteSpeculation passes in the restored nullable walker from a snapshot.

It looks like ```AnalyzeAndRewriteSpeculation``` uses different overload.

---
In reply to: [295550126](https://github.com/dotnet/roslyn/pull/36563#discussion_r295550126) [](ancestors = 295550126,295514274)